pandas/core/frame.py

```diff
@@ -5066,12 +5066,27 @@ def _replace_columnwise(
                 target, value = mapping[ax[i]]
                 newobj = ser.replace(target, value, regex=regex)
 
-                res.iloc[:, i] = newobj
+                res._isetitem(i, newobj)
 
         if inplace:
             return
         return res.__finalize__(self)
 
+    def _isetitem(self, loc: int, value):
+        cols = self.columns
+        if cols.is_unique:
+            col = cols[loc]
+            self[col] = value
+            return
+
+        # Otherwise we temporarily pin unique columns and call __setitem__
+        newcols = Index(range(len(cols)))
+        try:
+            self.columns = newcols
+            self[loc] = value
+        finally:
+            self.columns = cols
```

Comment:
+1 on this

Reply:
sure.

this is a fairly low priority, posted in the hopes it would help joris with ArrayManager indexing.  #40456 is the most blockery of the things I have going ATM.